From 68bbbfbc7beb69252ab8a389530a2dfb48fa9baa Mon Sep 17 00:00:00 2001
From: Jonas Gorski <jonas.gorski@gmail.com>
Date: Thu, 19 May 2022 10:09:04 +0200
Subject: [PATCH 11/47] kmodbuild.sh: don't treat undefined symbols as errors

Modern kernels treat undefined symbols as errors, but when building
modules individually, inter-module dependencies cannot be properly
resolved.

For now reduce the error to a warning to allow building with newer
kernels.

Upstream-Status: Inappropriate [should be fixed, not ignored]
Signed-off-by: Jonas Gorski <jonas.gorski@bisdn.de>
---
 tools/scripts/kmodbuild.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/scripts/kmodbuild.sh b/tools/scripts/kmodbuild.sh
index d5586b70fcc4..8fc07a6f405e 100755
--- a/tools/scripts/kmodbuild.sh
+++ b/tools/scripts/kmodbuild.sh
@@ -17,7 +17,7 @@ function build
     fi
     BUILD_DIR=$2
     INSTALL_DIR=$3
-    make -C $KERNEL M=$BUILD_DIR modules
+    make -C $KERNEL M=$BUILD_DIR KBUILD_MODPOST_WARN=1 modules
     make -C $KERNEL M=$BUILD_DIR INSTALL_MOD_PATH=`pwd` INSTALL_MOD_DIR="$3" modules_install
 }
 
-- 
2.51.0

